ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -490,10 +491,120 @@ module Vendor_nvidia = struct
   let whitelist_file () = !Xapi_globs.nvidia_whitelist
   let device_id_of_conf conf = conf.identifier.Identifier.pdev_id
 
+  (* A type to capture the XML tree with functions for use in Xmlm.input_doc_tree *)
```

Comment:
The tree type and associated functions look quite generic: I don't see anything about them that makes them belong in the Xapi_vgpu_type module. Could they go somewhere else, so we can re-use them from elsewhere in the future?

Reply:
They are quite generic indeed. I just needed them because the xmlm library is so basic. However, I'm not sure where else to put them. Not in an xmlmext.ml probably :)